### ISSUE TITLE
Hyper-parameter tuning

### ISSUE THREAD

**Issue by shpatrickguo** [shpatrickguo/datascience-projects#3]
https://medium.com/@pushkarmandot/https-medium-com-pushkarmandot-what-is-lightgbm-how-to-implement-it-how-to-fine-tune-the-parameters-60347819b7fc

**Comment by shpatrickguo** [comment]
Issues with hyper-parameter optimization:
Takes up way too much memory for my laptop.

Steps to amend this:
Data reading with memory footprint reduction. Implementation copied from [Eryk Lewson](https://towardsdatascience.com/make-working-with-large-dataframes-easier-at-least-for-your-memory-6f52b5f4b5c4)

Action Items:
Early stopping in LightGBM model training to avoid overtraining
Learning rate decay in LightGBM model training to improve convergence to the minimum
Hyperparameter optimisation of the model using random search in cross validation

**Comment by shpatrickguo** [comment]
Success: BayesSearchCV. 
Use Test as Eval 
Early stopping: 1000
CV=3

**Comment by shpatrickguo** [comment]
Issues with BayesSearchCV, crashes kernal on local host. iid Type error in Kaggle. 
Switch to RandomSearchCV
